File: moviejava/app/src/main/java/com/example/moviejava/data/remote/film/FilmService.java
package com.example.moviejava.data.remote.film;


import com.example.moviejava.data.remote.Constant;
import com.example.moviejava.model.casts.CastsFilm;
import com.example.moviejava.model.film.Film;
import com.example.moviejava.model.film.video.Video;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface FilmService {
    @GET("movie/{id}")
    Observable<Film> getFilm(@Path("id") int id);

    @GET("movie/{id}/videos")
    Observable<Video> getVideo(@Path("id") int id);

    @GET("movie/{id}/casts")
    Observable<CastsFilm> getCastFilm(@Path("id") int id);

}

File: movie/app/src/main/java/com/example/movie/data/remote/film/FilmService.kt
package com.example.movie.data.remote.film


import com.example.movie.data.remote.Constant
import com.example.movie.model.casts.CastsFilm
import com.example.movie.model.film.Film
import com.example.movie.model.film.video.Video
import retrofit2.http.GET
import retrofit2.http.Path

interface FilmService {
    @GET("movie/{id}")
    suspend fun getFilm(@Path("id") id: Int): Film

    @GET("movie/{id}/videos")
    suspend fun getVideo(@Path("id") id: Int): Video

    @GET("movie/{id}/casts")
    suspend fun getCastFilm(@Path("id") id: Int): CastsFilm

}